index.php

```diff
@@ -13,58 +13,63 @@
  * @author     Andreas Gohr <andi@splitbrain.org>
  */
-if(php_sapi_name() != 'cli-server') {
-    if(!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__).'/');
-    require_once(DOKU_INC.'inc/init.php');
+if (php_sapi_name() != 'cli-server') {
+    if (!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__) . '/');
+    require_once(DOKU_INC . 'inc/init.php');
 
-    send_redirect(DOKU_URL.'doku.php');
+    send_redirect(DOKU_URL . 'doku.php');
 }
 
-# ROUTER starts below
+// ROUTER starts below
 
-# avoid path traversal
+// avoid path traversal
 $_SERVER['SCRIPT_NAME'] = str_replace('/../', '/', $_SERVER['SCRIPT_NAME']);
 
-# routing aka. rewriting
-if(preg_match('/^\/_media\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # media dispatcher
+// routing aka. rewriting
+if (preg_match('/^\/_media\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // media dispatcher
     $_GET['media'] = $m[1];
     require $_SERVER['DOCUMENT_ROOT'] . '/lib/exe/fetch.php';
 
-} else if(preg_match('/^\/_detail\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # image detail view
+} elseif (preg_match('/^\/_detail\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // image detail view
     $_GET['media'] = $m[1];
     require $_SERVER['DOCUMENT_ROOT'] . '/lib/exe/detail.php';
 
-} else if(preg_match('/^\/_export\/([^\/]+)\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
-    # exports
+} elseif (preg_match('/^\/_export\/([^\/]+)\/(.*)/', $_SERVER['SCRIPT_NAME'], $m)) {
+    // exports
     $_GET['do'] = 'export_' . $m[1];
     $_GET['id'] = $m[2];
     require $_SERVER['DOCUMENT_ROOT'] . '/doku.php';
 
-} elseif($_SERVER['SCRIPT_NAME'] == '/index.php') {
-    # 404s are automatically mapped to index.php
-    if(isset($_SERVER['PATH_INFO'])) {
-        $_GET['id'] = $_SERVER['PATH_INFO'];
-    }
-    require $_SERVER['DOCUMENT_ROOT'] . '/doku.php';
+} elseif (
+    $_SERVER['SCRIPT_NAME'] !== '/index.php' &&
+    file_exists($_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'])
+) {
+    // existing files
 
-} else if(file_exists($_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'])) {
-    # existing files
-
-    # access limitiations
-    if(preg_match('/\/([\._]ht|README$|VERSION$|COPYING$)/', $_SERVER['SCRIPT_NAME']) or
+    // access limitiations
+    if (preg_match('/\/([._]ht|README$|VERSION$|COPYING$)/', $_SERVER['SCRIPT_NAME']) or
         preg_match('/^\/(data|conf|bin|inc)\//', $_SERVER['SCRIPT_NAME'])
     ) {
         die('Access denied');
     }
 
-    if(substr($_SERVER['SCRIPT_NAME'], -4) == '.php') {
+    if (substr($_SERVER['SCRIPT_NAME'], -4) == '.php') {
         # php scripts
         require $_SERVER['DOCUMENT_ROOT'] . $_SERVER['SCRIPT_NAME'];
     } else {
         # static files
         return false;
     }
+} else {
+    // treat everything else as a potential wiki page
+    // working around https://bugs.php.net/bug.php?id=61286
+    if (isset($_SERVER['PATH_INFO'])) {
+        $_GET['id'] = $_SERVER['PATH_INFO'];
+    } else {
```

Comment:
Regarding the discussion in https://github.com/splitbrain/dokuwiki/issues/3165#issuecomment-643070207, I would assume cli-server always has the root in DokuWiki root (because the previous `$_SERVER['SCRIPT_NAME'] !== '/index.php'` condition). Anyway it looks like the following patch works fine for `/`, `/index.php` (no argument set) and `/?id=index.php` (`index.php`), `index` (`index`).

```suggestion
    } elseif ($_SERVER['SCRIPT_NAME'] !== $_SERVER['PHP_SELF']) {
```

Reply:
Hum. I tried this. Unfortunately, with this, any page with a dot in the name result in :start being loaded.  For example `http://127.0.0.1:8000/truc:piquant.1.cmd` will serve :start. Other pages load ok.

Reply:
@schplurtz Thanks for testing! Proposing something new below.